### PR TITLE
Fix delete datastore

### PR DESCRIPTION
**Pull request by akarol**
{{pytest: -v -k 'test_delete_datastore'}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 823a5909411fddd4152aca5d62c7a36d008f3d2b:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/infrastructure/test_delete_infra_object.py [hunk 1]
```diff
@@ -109,7 +109,13 @@ def test_delete_datastore(setup_provider, provider_crud, remove_test):
...
     vm_count = test_datastore.get_vms()
-    if len(host_count) == 0 and len(vm_count) == 0:
+    if len(host_count) != 0 and len(vm_count) == 0:
+        test_datastore.delete_all_attached_hosts()
+        test_datastore.wait_for_delete_all()
+        test_datastore.delete(cancel=False)
+    elif len(host_count) == 0 and len(vm_count) != 0:
+        test_datastore.delete_all_attached_vms()
+        test_datastore.wait_for_delete_all()
         test_datastore.delete(cancel=False)
```
Comment (psav) [inline cfme/tests/infrastructure/test_delete_infra_object.py]: Couldn't lines 112-125 be written like so?
```python
if len(host_count) != 0:
     test_datastore.delete_all_attached_hosts()
     test_datastore.wait_for_delete_all()
if len(vm_count) != 0:
     test_datastore.delete_all_attached_vms()
     test_datastore.wait_for_delete_all()
test_datastore.delete(cancel=False)
```

Comment (akarol) [inline cfme/tests/infrastructure/test_delete_infra_object.py]: @psav Done!

**Comment by dajoRH** [comment]
Lint report for commit ca4a9e4e2fcbf87911e1e326070abe4a2225db05:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/infrastructure/test_delete_infra_object.py [hunk 2]
```diff
@@ -109,14 +109,13 @@ def test_delete_datastore(setup_provider, provider_crud, remove_test):
     test_datastore = datastore.Datastore(name=data_store)
     host_count = test_datastore.get_hosts()
     vm_count = test_datastore.get_vms()
-    if len(host_count) == 0 and len(vm_count) == 0:
-        test_datastore.delete(cancel=False)
-    else:
+    if len(host_count) != 0:
```
Comment (seandst) [inline cfme/tests/infrastructure/test_delete_infra_object.py]: sequences with length > 0 evaluate as True; This and line 115 should be `if host_count:`

**Comment by dajoRH** [comment]
Lint report for commit c430449562c385420319cf65dc93b6c968e8fe58:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5554cb1bcaf6827910216b6fa20de7c3863d83be:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit cd9c28fedc25b910c6d6904f4da4ead0317c7ed6:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d1130aee8293287d699123d11c82bf2ab61b3020:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**